cmd/federation-controller/main.go

```diff
@@ -1,7 +1,55 @@
 package main
 
-import "os"
+import (
+	"flag"
+	"fmt"
+	"os"
+	"github.com/jewertow/federation/internal/pkg/config"
+	"gopkg.in/yaml.v3"
+)
+
+// Global variable to store the parsed arguments and "flag" arguments
+var (
+	AppConfig config.Federation
```

Comment:
We shouldn't export this variable, because we will instantiate dependent components in this file, like exported service set watcher, and then we won't be able to import this config variable, because that would cause circular dependency.
```suggestion
```

Reply:
Remove from export